### PR TITLE
Change oak_main signature to take a handle param

### PR DESCRIPTION
**Pull request by daviddrysdale**
Not yet used, in preparation for a future commit.

### CODE REVIEW

File: docs/abi.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Two specific sets of integer values are used in multiple places in the ABI:
 Each Oak Module must expose the following **exported function** as a
 [WebAssembly export](https://webassembly.github.io/spec/core/syntax/modules.html#exports):
 
-- `oak_main: () -> u32`: Invoked when the Oak Manager executes the Oak Node.
+- `oak_main: (u64) -> u32`: Invoked when the Oak Manager executes the Oak Node.
   This function should perform its own event loop, reading incoming messages
```
Comment (tiziano88) [inline docs/abi.md]: Mention what the parameter is in this paragraph.

File: sdk/rust/oak_tests/src/lib.rs [hunk 2]
```diff
@@ -789,7 +789,7 @@ pub fn start<S: ::std::hash::BuildHasher>(
         let node_name = name.clone();
         let thread_handle = spawn(move || {
             set_node_name(node_name);
-            entrypoint()
+            entrypoint(0)
```
Comment (tiziano88) [inline sdk/rust/oak_tests/src/lib.rs]: Is `0` meant to mean "invalid handle", or is it just an arbitrary handle number that is irrelevant for the rest of the logic?

File: docs/abi.md (hunk 1 above)
Comment (daviddrysdale) [reply docs/abi.md]: Done.

File: sdk/rust/oak_tests/src/lib.rs (hunk 2 above)
Comment (daviddrysdale) [reply sdk/rust/oak_tests/src/lib.rs]: It's invalid handle (for now); added a constant and a comment.

**Pull request merged**